modules/custom/dkan_data/src/ValueReferencer.php

```diff
@@ -206,16 +234,40 @@ protected function createPropertyReference(string $property_id, $value) {
     return $node->uuid();
   }
 
+  /**
+   *
+   *
+   * @param array $params
+   */
+  protected function setDereferenceMethod(array $params) {
+    if (isset($params['values']) && $params['values'] == 'both') {
+      return $this->dereferenceMethod = self::DEREFERENCE_OUTPUT_BOTH;
+    }
+    elseif (isset($params['values']) && $params['values'] == 'identifier') {
+      return $this->dereferenceMethod = self::DEREFERENCE_OUTPUT_IDENTIFIER;
+    }
+    else {
+      return $this->dereferenceMethod = self::DEREFERENCE_OUTPUT_DATA;
+    }
+  }
+
   /**
    * Replaces value references in a dataset with with their actual values.
    *
    * @param \stdClass $data
    *   The json metadata object.
+   * @param array $params
+   *   An array with the http request's query parameters, if any.
    *
    * @return mixed
    *   Modified json metadata object.
    */
-  public function dereference(stdClass $data) {
+  public function dereference(stdClass $data, array $params = []) {
```

Comment:
@thierrydallacroce would it be better if we expect one of our constants to be used as the means to set the dereferencing style we are using. Right now we are coupling stuff in here to a fairly arbitrary array "params".

Reply:
@fmizzell agreed, it will make it better.

Reply:
Modified code per your recommendation.